tests/upgrade/0.8.2-0.9.0/jiva/suite_test.go

```diff
@@ -0,0 +1,257 @@
+
+	By("verifying status as bound")
+	status := ops.IsPVCBound(pvcName)
+	Expect(status).To(Equal(true), "while checking status equal to bound")
+
+	// TODO
+	// By("applying openebs 0.9.0")
+	// applyYAMLFromURL("https://openebs.github.io/charts/openebs-operator-0.9.0-RC3.yaml", "")
+
+})
+
+var _ = AfterSuite(func() {
+
+	By("deleting above pvc")
+	err := ops.PVCClient.Delete(pvcName, &metav1.DeleteOptions{})
+	Expect(err).To(
+		BeNil(),
+		"while deleting pvc {%s} in namespace {%s}",
+		pvcName,
+		nsName,
+	)
+
+	By("verifying controller pod count")
+	controllerPodCount := ops.GetPodRunningCountEventually(nsName, ctrlLabel, 0)
+	Expect(controllerPodCount).To(Equal(0), "while checking controller pod count")
+
+	By("verifying replica pod count")
+	replicaPodCount := ops.GetPodRunningCountEventually(nsName, replicaLabel, 0)
+	Expect(replicaPodCount).To(Equal(0), "while checking replica pod count")
+
+	By("deleting storageclass")
+	err = ops.SCClient.Delete(scName, &metav1.DeleteOptions{})
+	Expect(err).To(BeNil(), "while deleting storageclass {%s}", scObj.Name)
+
+	By("Cleanup")
+	deleteArtifact(jobArtifact, "job")
+	deleteArtifact(runtaskArtifact, "")
+	deleteArtifact(crArtifact, "")
+	deleteArtifact(rbacArtifact, "")
+	deleteArtifact(openebsArtifact, "")
+	podList, err := ops.PodClient.
+		WithNamespace("default").
+		List(metav1.ListOptions{})
+	Expect(err).ShouldNot(HaveOccurred())
+	for _, po := range podList.Items {
+		if po.Status.Phase == "Succeeded" {
+			err = ops.PodClient.Delete(po.Name, &metav1.DeleteOptions{})
+			Expect(err).To(BeNil(), "while deleting completed pods")
+		}
+	}
+})
+
+func getArtifactFromURL(url string) artifacts.Artifact {
+	// Read yaml file from the url
+	o, err := fetch(url)
+	Expect(err).ShouldNot(HaveOccurred())
+	defer o.Close()
+	b, err := ioutil.ReadAll(o)
+	Expect(err).ShouldNot(HaveOccurred())
+	// Connvert the yaml to unstructured objects
+	yamlString := string(b)
+	return artifacts.Artifact(yamlString)
+}
+
+func applyArtifact(artifact artifacts.Artifact, flag string) {
+	artifactList, errs := artifacts.GetArtifactsListUnstructured(artifact)
+	Expect(errs).Should(HaveLen(0))
+	// Applying unstructured objects
+	for i, artifact := range artifactList {
+		if flag == "job" && i == 0 {
+			unstructured.SetNestedStringMap(artifact.Object, data, "data")
+		}
+		if flag == "job" && i == 1 {
+			artifact.SetNamespace("default")
+		}
+		cu := k8s.CreateOrUpdate(
+			k8s.GroupVersionResourceFromGVK(artifact),
+			artifact.GetNamespace(),
+		)
+		_, err := cu.Apply(artifact)
+		Expect(err).ShouldNot(HaveOccurred())
+	}
+}
+
+func deleteArtifact(artifact artifacts.Artifact, flag string) {
+	artifactList, errs := artifacts.GetArtifactsListUnstructured(artifact)
+	Expect(errs).Should(HaveLen(0))
+	// Applying unstructured objects
+	for i, artifact := range artifactList {
+		if flag == "job" && i == 1 {
+			artifact.SetNamespace("default")
+		}
+		del := k8s.DeleteResource(
```

Comment:
Please check if `https://github.com/openebs/maya/tree/master/pkg/unstruct/v1alpha2` can handle this?

Reply:
Done